frontend/model/notifications/vuexModule.js

```diff
@@ -1,19 +1,31 @@
 'use strict'
 
 import { Vue } from '@common/common.js'
-import type { Notification } from './types.flow.js'
-import './selectors.js'
-import { compareOnTimestamp, isNew, isOlder } from './utils.js'
-import * as keys from './mutationKeys.js'
 import { cloneDeep } from '~/frontend/model/contracts/shared/giLodash.js'
+import * as keys from './mutationKeys.js'
+import './selectors.js'
+import { MAX_AGE_READ, MAX_AGE_UNREAD } from './storageConstants.js'
+import type { Notification } from './types.flow.js'
+import { age, compareOnTimestamp, isNew, isOlder } from './utils.js'
 
 const defaultState = {
   items: [], status: {}
 }
 
 const getters = {
   notifications (state, getters, rootState) {
-    return state.items.map(item => ({ ...item, ...state.status[item.hash] }))
+    return state.items.map(item => {
+      const notification = { ...item, ...state.status[item.hash] }
+      // Notifications older than MAX_AGE_UNREAD are discarded
+      if (age(notification) > MAX_AGE_UNREAD) {
+        return null
```

Comment:
OK, they are discarded in the getter, but where are they discarded from the state?

Reply:
They're discarded from the state when the state is saved (see `state/vuex/save`)

Reply:
I genuinely do not see where that happens. Can you point it out?

Reply:
The line is literally part of this commit (although the change in question is related but actually a bugfix for a different issue).

https://github.com/okTurtles/group-income/pull/2333/files#diff-170fa1f8952aa138da06e2d0a0e0039fffda8cdf4168989b4886eceaf66d502cR126

As you can see, each time the state is saved, the notifications are filtered.

Reply:
Then my logic comprehension might be broken. I saw that line and that line doesn't appear to do be related to these changes here as it is accessing the notifications through the `state` (which is unfiltered) unlike this getter. What am I missing?

Reply:
> What am I missing?

You're missing it here:

>  (which is unfiltered)

The call to `applyStorageRules` does the filtering. Hence, the only the `getter` needs to be updated.

Reply:
OK, I think I see what you mean, it's this line:

```js
let items = notifications.filter(item => !isExpired({ ...item, ...status[item.hash] }))
```

And then the `status[item.hash]` supposedly has the correct value to get `maxAge` to return `MAX_AGE_UNREAD`, is my guess. Although per my [other question](https://github.com/okTurtles/group-income/pull/2333#discussion_r1747839448) I'm not super clear on how that would have the correct value, but in my tests this seemed to work so I'll go ahead and approve once the stray debug logging is removed. If there are problems we can return to this later.

Reply:
Well, I'm not sure what's confusing to you, but I can comment on what was somewhat confusing to me.

There are two different things going on: saving or mutating the UI and display concerns. In addition, there notification state is split up into the `items` and `status` properties. `items` is generated when a notification happens, and `status` mostly (only?) holds the read / unread status.

The 'saving' part is mostly addressed in this PR by adjusting the constants. That was mostly already working (the change I pointed out in `state.js` didn't entirely work because it looks at the `read` property, but it wasn't gonna find it because it's usually undefined in `items`; the change makes it also look into `status`). In short, saving already filters things out based on age.

This PR addresses the UI part. This has two goals:

1. Since the `read` state for notifications that are too old (i.e., between the two windows) will _not_ be saved, it sets `read` to true for display purposes.
2. It also filters out notifications that are too old. This is for consistency, but it's not that important since saving the state takes care of removing these old notifications.